### PR TITLE
[dynamo][functional_collectives] Add all_to_all_single, all_gather_list, reduce_scatter_list to dynamo remapping

### PR DESCRIPTION
**Pull request by IvanKobzarev**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #119683

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225

Differential Revision: [D53758434](https://our.internmc.facebook.com/intern/diff/D53758434)

### CODE REVIEW

**Review by ezyang — APPROVED** [review]
Needs a test, but looks good

**Comment by IvanKobzarev** [comment]
@IvanKobzarev has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D53758434).

**Comment by IvanKobzarev** [comment]
@IvanKobzarev has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D53758434).

File: test/distributed/test_inductor_collectives.py [hunk 1]
```diff
@@ -814,9 +811,6 @@ def func(inp, out, *, pg):
         compiled(inputs, outputs, pg=GroupMember.WORLD)
         func(inputs, correct_outputs, pg=GroupMember.WORLD)
         assert counter.frame_count == 1
-
```
Comment (wconstab) [inline test/distributed/test_inductor_collectives.py]: why is this being deleted?

**Pull request closed without merging**